Shared/messages/MessagesOverlay.cpp

```diff
@@ -37,153 +37,59 @@ namespace Dsa {
-  m_dynamicEntityLayer->setName(messageType);
-  SceneView* scene = static_cast<SceneView*>(m_geoView);
-  scene->arcGISScene()->operationalLayers()->append(m_dynamicEntityLayer);
-}
-
-MessagesOverlay::~MessagesOverlay()
-{
-}
-
-/*!
-  \brief Returns the Esri:ArcGISRuntime::Renderer in use by the overlay.
- */
-Renderer* MessagesOverlay::renderer() const
-{
-  return m_renderer.data();
-}
-
-/*!
-  \brief Sets the Esri:ArcGISRuntime::Renderer for the overlay to \a renderer.
- */
-void MessagesOverlay::setRenderer(Renderer* renderer)
-{
-  if (m_renderer == renderer)
-    return;
-
-  m_renderer = renderer;
-
-  m_dynamicEntityLayer->setRenderer(m_renderer);
+  this->setLayerId(messageType);
+  this->setName(messageType);
+  messageFeed->setMessagesOverlay(this);
 }
 
 /*!
-  \brief Returns the Esri:ArcGISRuntime::SurfacePlacement for the overlay.
- */
-SurfacePlacement MessagesOverlay::surfacePlacement() const
-{
-  return m_surfacePlacement;
-}
-
-/*!
-  \brief Sets the Esri:ArcGISRuntime::SurfacePlacement for the overlay to \a surfacePlacement.
+  \brief Destructor.
  */
-void MessagesOverlay::setSurfacePlacement(SurfacePlacement surfacePlacement)
-{
-  if (m_surfacePlacement == surfacePlacement)
-    return;
-
-  m_surfacePlacement = surfacePlacement;
-
-  m_dynamicEntityLayer->setSceneProperties(LayerSceneProperties(m_surfacePlacement));
-}
+MessagesOverlay::~MessagesOverlay() = default;
 
 /*!
   \brief Returns the message type for the overlay.
  */
 QString MessagesOverlay::messageType() const
 {
-  return m_dynamicEntityLayer->layerId();
+  return this->layerId();
 }
 
 /*!
   \brief Sets the message type for the overlay to \a messageType.
  */
 void MessagesOverlay::setMessageType(const QString& messageType)
 {
-  if (m_dynamicEntityLayer->layerId() == messageType)
+  if (this->layerId() == messageType)
     return;
 
-  m_dynamicEntityLayer->setLayerId(messageType);
-}
-
-/*!
-  \brief Returns the Esri:ArcGISRuntime::GraphicsOverlay object for the overlay.
- */
-DynamicEntityLayer* MessagesOverlay::dynamicEntityLayer() const
-{
-  return m_dynamicEntityLayer;
-}
-
-/*!
-  \brief Returns the Esri:ArcGISRuntime::GeoView for the overlay.
- */
-GeoView* MessagesOverlay::geoView() const
-{
-  return m_geoView;
+  this->setLayerId(messageType);
 }
 
-/*!
-  \brief Returns whether the overlay is visible.
- */
-bool MessagesOverlay::isVisible() const
+DynamicEntity* MessagesOverlay::getDynamicEntityById(quint64 entityId)
 {
-  return m_dynamicEntityLayer->isVisible();
+  return m_messageFeed->getDynamicEntityById(entityId);
 }
 
-/*!
-  \brief Sets the overlay to be \a visible.
- */
-void MessagesOverlay::setVisible(bool visible)
+const QHash<quint64, Esri::ArcGISRuntime::DynamicEntity*>& MessagesOverlay::dynamicEntities()
 {
-  if (m_dynamicEntityLayer->isVisible() == visible)
-    return;
-
-  m_dynamicEntityLayer->setVisible(visible);
-
-  emit visibleChanged();
+  return m_messageFeed->dynamicEntities();
```

Comment:
Double check this is ok. Returning a const reference is typically done by returning a member directly. This looks suspicious (but might be fine).